base/boot.jl

```diff
@@ -332,6 +333,39 @@ end
 atdoc     = (str, expr) -> Expr(:escape, expr)
 atdoc!(λ) = global atdoc = λ
 
+
+# simple stand-alone print definitions for debugging
+abstract IO
+type CoreSTDOUT <: IO end
+type CoreSTDERR <: IO end
+const STDOUT = CoreSTDOUT()
+const STDERR = CoreSTDERR()
+io_pointer(::CoreSTDOUT) = Intrinsics.pointerref(Intrinsics.cglobal(:jl_uv_stdout, Ptr{Void}), 1)
+io_pointer(::CoreSTDERR) = Intrinsics.pointerref(Intrinsics.cglobal(:jl_uv_stderr, Ptr{Void}), 1)
+
+unsafe_write(io::IO, x::Ptr{UInt8}, nb::UInt) =
+    ccall(:jl_uv_puts, Void, (Ptr{Void}, Ptr{UInt8}, UInt), io_pointer(io), x, nb)
+unsafe_write(io::IO, x::Ptr{UInt8}, nb::Int) =
+    ccall(:jl_uv_puts, Void, (Ptr{Void}, Ptr{UInt8}, Int), io_pointer(io), x, nb)
+write(io::IO, x::UInt8) =
+    ccall(:jl_uv_putb, Void, (Ptr{Void}, UInt8), io_pointer(io), x)
+write(io::IO, x::String) =
+    unsafe_write(io, ccall(:jl_array_ptr, Ptr{UInt8}, (Any,), x.data), sizeof(x.data))
+
+show(io::IO, x::ANY) = ccall(:jl_static_show, Void, (Ptr{Void}, Any), io_pointer(io), x)
+print(io::IO, x::Char) = ccall(:jl_uv_putc, Void, (Ptr{Void}, Char), io_pointer(io), x)
+print(io::IO, x::String) = write(io, x)
+print(io::IO, x::ANY) = show(io, x)
+print(io::IO, a::ANY...) = for x in a; print(io, x); end
+println(io::IO) = write(io, 0x0a) # 0x0a = '\n'
+println(io::IO, x::ANY...) = (print(io, x...); println(io))
+
+println() = println(STDOUT)
+println(a::ANY) = println(STDOUT, a)
+print(a::ANY) = print(STDOUT, a)
+show(a::ANY) = show(STDOUT, a)
```

Comment:
yes, fixed (also fixed the function, which previously required features not available to Core)